File: DeepSpace-Java/src/deepspace/PowerEfficientSpaceStation.java
package deepspace;

public class PowerEfficientSpaceStation extends SpaceStation {
    private float EFFICIENCYFACTOR = 1.10f;

    PowerEfficientSpaceStation (SpaceStation estacion) {
        super(estacion);
    }

    @Override
    public float fire () {
        return super.fire() * EFFICIENCYFACTOR;
    }

    @Override
    public float protection () {
        return super.protection() * EFFICIENCYFACTOR;
    }

    public Transformation setLoot (Loot loot) {
        super.setLoot(loot);

        return loot.getEfficient()
            ?   Transformation.GETEFFICIENT
            :   Transformation.NOTRANSFORM;
    }

    public PowerEfficientSpaceStationToUI getUIversion () {
        return new PowerEfficientSpaceStationToUI(this);
    }

    @Override
    public String toString() {
        return getUIversion().toString();
    }
}

File: DeepSpace-Ruby/lib/PowerEfficientSpaceStation.rb
#encoding:utf-8

require_relative "Transformation"
require_relative "SpaceStation"
require_relative "PowerEfficientSpaceStationToUI"

module Deepspace
    class PowerEfficientSpaceStation < SpaceStation
        @@EFFICIENCYFACTOR = 1.1

        def initialize (station)
            createStation(station)
        end

        def fire
            disparo = super*@@EFFICIENCYFACTOR
        end

        def protection
            proteccion = super*@@EFFICIENCYFACTOR
        end

        def setLoot(s)
            super
            if s.getefficient
                Transformation::GETEFFICIENT
            else
                Transformation::NOTRANSFORM
            end
        end

        def to_s
            "Power Efficient Station Space Station"+super
        end
    end

end
